### DISCUSSION TITLE
Issues running Tutorials and Examples

### DISCUSSION THREAD

**Discussion by camposgvk** [IDAES/idaes-pse#1031] [General]
Hi all, I just installed IDAES (advanced user installation) and tried to run some of the tutorials and examples.
I am having a few errors all over the place, so I am starting to suspect there might be something wrong with the installation.
Some examples of errors:

1) code: "m.fs = FlowsheetBlock(default={"dynamic": False})" 
error: "ValueError: key 'default' not defined for ConfigDict '' and implicit (undefined) keys are not allowed""

2) code: "from idaes.generic_models.properties.activity_coeff_models.BTX_activity_coeff_VLE import BTXParameterBlock"
error: "ModuleNotFoundError: No module named 'idaes.generic_models'"

3) code: "run "notebook_test_script.py"" (from introduction tutorial)
error: (same error as (1) above)

I can circumvent some of these errors (for example, it seems like the folder generic_models was renamed as models), but I wanted to double check if other people have had the same issues or if it is some local installation errors I am getting.

Thank you.

**Comment by andrewlee94** [comment]
Looking at the errors you are receiving, it appears there is a mismatch between the version of IDAES you have and the examples. IDAES is in the process of transitioning to v2.0 which involves a number of changes in the API thus most of the old examples will no longer work.

From what you said, I suspect you installed the latest `main` version of `idaes-pse`, but still have an older version of the examples (probably from v1.13 which was the last stable release). If you install the latest version of the examples (see https://idaes-pse.readthedocs.io/en/stable/reference_guides/commands/get_examples.html), this should resolve your issues.

**Comment by camposgvk** [comment]
Thanks @andrewlee94 

For those having the same issue:

Link for the get-examples usage page: https://idaes-pse.readthedocs.io/en/stable/reference_guides/commands/get_examples.html
(some of the commands didn't work for me, had to change them slightly. Running on Windows with anaconda prompt)

1) List all versions available: "idaes get-examples -l -U"
(need to use the unstable suffix -U to show versions compatible with idaes >2.0)
2) Install latest version: "idaes get-examples -U --version 2.0.0a3"

**Reply by camposgvk** [reply]
Unfortunately, still having the same error from the "default" argument in:
`m.fs = FlowsheetBlock(default={"dynamic": False})`
Not sure if the get-examples bypassed my options and retrieved an earlier version.

**Reply by andrewlee94** [reply]
I think the issue is that you need something even more recent than 2.0.0a3 when running with the latest version of `idaes-pse` but I am not sure how you can get it through the `get_examples` command.

There are a few possible solutions:

1. Download or clone the main branch of `idaes/examples-pse` repo directly.
2. The next release of IDAES will occur in a couple of weeks-  if you can wait that long then there should be a new version of the examples released which should resolve this.

**Reply by camposgvk** [reply]
Thanks. Cloning the examples-pse GitHub repo seems to solve most of my issues.
A few observations for those having the same issue:
- I cloned the examples-pse repo directly inside of my idaes-pse forked/cloned repo and it works well (not sure if this is not best practice).
- The "Examples" and "Tutorials" folders are inside examples-pse\src
- you can install the idaes-examples package running "python setup.py develop" from anaconda prompt (if you plan to change the package and have it update automatically, instead use "python setup.py install")
- the "run "notebook_test_script.py"" code still fails for the "IDAES Import Checks" step.
